aspnetcore/release-notes/aspnetcore-3.0.md

```diff
@@ -0,0 +1,541 @@
+* The ASP.NET Core 3.0 version is shown second. The green boxes indicate code that has been added. The 3.0 version requires `using Microsoft.Extensions.Hosting;`.
+
+## Host configuration and options
+
+Host builder configuration is provided by <xref:Microsoft.Extensions.Hosting.HostBuilder.ConfigureHostConfiguration*>:
+
+```csharp
+public static IHostBuilder CreateHostBuilder(string[] args) =>
+    Host.CreateDefaultBuilder(args)
+        .ConfigureHostConfiguration(configHost =>
+        {
+            configHost.SetBasePath(Directory.GetCurrentDirectory());
+            configHost.AddJsonFile("hostsettings.json", optional: true);
+            configHost.AddEnvironmentVariables(prefix: "PREFIX_");
+            configHost.AddCommandLine(args);
+    });
+```
+
+Prior to the release of ASP.NET Core 3.0, environment variables prefixed with `ASPNETCORE_` were loaded for host configuration of the Web Host. In 3.0, `AddEnvironmentVariables` is used to load environment variables prefixed with `DOTNET_` for host configuration with `CreateDefaultBuilder`. 
+
+To configure host options in code, call `Configure<HostOptions>` on the host builder's services provided by `ConfigureServices`:
+
+```csharp
+public static IHostBuilder CreateHostBuilder2(string[] args) =>
+    Host.CreateDefaultBuilder(args)
+        .ConfigureServices((hostContext, services) =>
+        {
+            services.Configure(option =>
+            {
+                option.ShutdownTimeout = System.TimeSpan.FromSeconds(20);
+            });
+        });
+```
+
+As a result of this change, the Startup class now only supports *constructor injection* of the following types: `IHostEnvironment`, `IWebHostEnvironment`, and `IConfiguration`. All services can still be injected directly as arguments to the `Startup.Configure` method. For more information, see [aspnet/Announcements#353](https://github.com/aspnet/Announcements/issues/353).
+
+For more information, see the following articles:
+
+* <xref:fundamentals/configuration/index>
+* <xref:fundamentals/host/generic-host>
+
+## Kestrel
+
+* Kestrel configuration has been updated for the migration to the Generic Host. In 3.0, Kestrel is configured on the web host builder provided by `ConfigureWebHostDefaults`.
+* Connection Adapters have been removed from Kestrel and replaced with Connection Middleware, which is similar to HTTP Middleware in the ASP.NET Core pipeline but for lower-level connections.
+* The Kestrel transport layer has been exposed as a public interface in `Connections.Abstractions`.
+* Ambiguity between headers and trailers has been resolved by moving trailing headers to a new collection.
+* Synchronous IO APIs, such as `HttpReqeuest.Body.Read`, are a common source of thread starvation leading to app crashes. In 3.0, `AllowSynchronousIO` is disabled by default.
+
+For more information, see <xref:migration/22-to-30>.
+
+## Hosting EventSourceProvider new request counters
```

Comment:
```suggestion
## Request counters
```

Let's remove reference to the event source altogether in the heading. At a minimum, it would need to be `EventSource`, not `EventSourceProvider` (as below 😉). I don't see a need to call out event source at all though. These are counters related to requests, that's sufficient description for a header IMO.